openedx/core/djangoapps/user_api/preferences/tests/test_api.py

```diff
@@ -1,61 +1,41 @@
 # -*- coding: utf-8 -*-
-""" Tests for the profile API. """
-from django.contrib.auth.models import User
-
+"""
+Unit tests for preference APIs.
+"""
 import ddt
+
 from django.test.utils import override_settings
 from nose.tools import raises
+import datetime
 from dateutil.parser import parse as parse_datetime
 from pytz import UTC
-from xmodule.modulestore.tests.factories import CourseFactory
+from django.contrib.auth.models import User
+
 from xmodule.modulestore.tests.django_utils import ModuleStoreTestCase
-import datetime
+from xmodule.modulestore.tests.factories import CourseFactory
 
-from ..accounts.api import get_account_settings
-from ..api import account as account_api
-from ..api import profile as profile_api
-from ..models import UserProfile, UserOrgTag
+from openedx.core.djangoapps.user_api.accounts.api import create_account
+from openedx.core.djangoapps.user_api.models import UserProfile, UserOrgTag
+from openedx.core.djangoapps.user_api.preferences.api import (
+    update_email_opt_in, set_user_preference, get_user_preferences, update_user_preferences
+)
 
 
 @ddt.ddt
-class ProfileApiTest(ModuleStoreTestCase):
+class UpdateEmailOptInTests(ModuleStoreTestCase):
 
     USERNAME = u'frank-underwood'
     PASSWORD = u'ṕáśśẃőŕd'
     EMAIL = u'frank+underwood@example.com'
 
-    def test_create_profile(self):
-        # Create a new account, which should have an empty profile by default.
-        account_api.create_account(self.USERNAME, self.PASSWORD, self.EMAIL)
-
-        # Retrieve the account settings
-        user = User.objects.get(username=self.USERNAME)
-        account_settings = get_account_settings(user)
-
-        # Expect a date joined field but remove it to simplify the following comparison
-        self.assertIsNotNone(account_settings['date_joined'])
-        del account_settings['date_joined']
-
-        # Expect all the values to be defaulted
-        self.assertEqual(account_settings, {
-            'username': self.USERNAME,
-            'email': self.EMAIL,
-            'name': u'',
-            'gender': None,
-            'language': u'',
-            'goals': None,
-            'level_of_education': None,
-            'mailing_address': None,
-            'year_of_birth': None,
-            'country': None,
-        })
-
     def test_update_and_retrieve_preference_info(self):
-        account_api.create_account(self.USERNAME, self.PASSWORD, self.EMAIL)
+        # TODO: move test into preferences API test.
```

Comment:
@andy-armstrong There are a couple of test points here that should be covered by your new API tests, and then these can be deleted (test_update_and_retrieve_preference_info and test_update_and_retrieve_preference_info_unicode).